clang/test/BoundsSafety/Sema/unsafe-buffer-usage-interop-crash.cpp

```diff
@@ -11,12 +11,15 @@ class MyClass {
 namespace value_dependent_assertion_violation {
   // Running attribute-only mode on the C++ code below crashes
   // previously.
-  void g(unsigned);
+  void g(int * __counted_by(n) * p, size_t n);
 
   template<typename T>
   struct S {
     void f(T p) {
-      g(sizeof(p));
+      g(nullptr, sizeof(p));
```

Comment:
Are we certain this still exercises the same path as caused the crash previously, or should we keep the old function and add another?

Reply:
The old one was testing for no type checking in a dependent context, if there is no bounds attribute.  Now we are testing that the same path will not be reached under a stronger condition, i.e., no type checking in a dependent context.

So the new test is stronger and covers the old one after the change of this PR.